app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java

```diff
@@ -494,35 +506,25 @@ void setMessage(NotificationViewData.Concrete notificationViewData, LinkListener
             switch (type) {
                 default:
                 case FAVOURITE: {
-                    icon = ContextCompat.getDrawable(context, R.drawable.ic_star_24dp);
-                    if (icon != null) {
-                        icon.setColorFilter(ContextCompat.getColor(context,
-                                R.color.tusky_orange), PorterDuff.Mode.SRC_ATOP);
-                    }
-
+                    icon = getIconWithColor(context, R.drawable.ic_star_24dp, R.color.tusky_orange);
                     format = context.getString(R.string.notification_favourite_format);
                     break;
                 }
                 case REBLOG: {
-                    icon = ContextCompat.getDrawable(context, R.drawable.ic_repeat_24dp);
-                    if (icon != null) {
-                        icon.setColorFilter(ContextCompat.getColor(context,
-                                R.color.tusky_blue), PorterDuff.Mode.SRC_ATOP);
-                    }
-
+                    icon = getIconWithColor(context, R.drawable.ic_repeat_24dp, R.color.tusky_blue);
                     format = context.getString(R.string.notification_reblog_format);
                     break;
                 }
                 case STATUS: {
-                    icon = ContextCompat.getDrawable(context, R.drawable.ic_home_24dp);
-                    if (icon != null) {
-                        icon.setColorFilter(ContextCompat.getColor(context,
-                                R.color.tusky_blue), PorterDuff.Mode.SRC_ATOP);
-                    }
-
+                    icon = getIconWithColor(context, R.drawable.ic_home_24dp, R.color.tusky_blue);
                     format = context.getString(R.string.notification_subscription_format);
                     break;
                 }
+                case UPDATE: {
+                    icon = getIconWithColor(context, R.drawable.ic_home_24dp, R.color.tusky_blue);
```

Comment:
I have never seen an edit notification in the webinterface, but shouldn't the icon be an edit icon, like a pencil?

Reply:
Good point, that is what it is in the web ui